### ISSUE TITLE
Reconsider `is_*` check for `wrong_self_convention`?

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust-clippy#8513]
### Summary

`wrong_self_convention` should allow `&mut self` in addition to `&self` for `is_*` functions.

### Lint Name

wrong_self_convention

### Reproducer

I tried this code:

```rust
use async_compat::Compat;
use tokio::net::UnixStream;
use memcache_async::ascii::Protocol;

struct Connector(Protocol<Compat<UnixStream>>);

impl Connector {
    async fn is_valid(&mut self) -> std::io::Result<()> {
        self.0.version().await.map(drop)
    }   
}
```

I saw this happen:

```
warning: methods called `is_*` usually take `self` by reference or no `self`
   --> src/main.rs:102:23
    |
102 |     async fn is_valid(&mut self) -> std::io::Result<()> {
    |                       ^^^^^^^^^
    |
    = help: consider choosing a less ambiguous name
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention
```

I expected to see this happen: No warning. Warning here seems very strange - the name of the function is clear, and it's behaving exactly as intended. I think the lint should allow `&mut self` in addition to `&self`.

See also https://github.com/rust-lang/rust-clippy/issues/8480.

### Version

```text
rustc 1.60.0-nightly (b17226fcc 2022-02-18)
binary: rustc
commit-hash: b17226fcc11587fed612631be372a5b4cb89988a
commit-date: 2022-02-18
host: x86_64-unknown-linux-gnu
release: 1.60.0-nightly
LLVM version: 14.0.0
```

**Comment by Alexendoo** [comment]
fixed by #8738
